sdk/identity/identity/review/identity-api-review.md

```diff
@@ -0,0 +1,137 @@
+    }
+
+// @public
+export type BrowserLoginStyle = "redirect" | "popup";
+
+// @public
+export class ChainedTokenCredential implements TokenCredential {
+    constructor(...sources: TokenCredential[]);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export class ClientCertificateCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, certificatePath: string, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export class ClientSecretCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, clientSecret: string, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export class DefaultAzureCredential extends ChainedTokenCredential {
+    constructor(identityClientOptions?: IdentityClientOptions);
+}
+
+// @public
+export class DeviceCodeCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, userPromptCallback: DeviceCodePromptCallback, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export interface DeviceCodeDetails {
+    message: string;
+    userCode: string;
+    verificationUri: string;
+}
+
+// @public
+export type DeviceCodePromptCallback = (deviceCodeDetails: DeviceCodeDetails) => void;
+
+// @public
+export class EnvironmentCredential implements TokenCredential {
+    constructor(options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+}
+
+// @public
+export interface ErrorResponse {
+    correlation_id?: string;
+    error: string;
+    error_codes?: number[];
+    error_description: string;
+    timestamp?: string;
+    trace_id?: string;
+}
+
+// @public
+export function getDefaultAzureCredential(): TokenCredential;
+
+export { GetTokenOptions }
+
+// @public
+export interface IdentityClientOptions extends ServiceClientOptions {
+    authorityHost?: string;
+}
+
+// @public
+export class InteractiveBrowserCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, options?: InteractiveBrowserCredentialOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+}
+
+// @public
+export interface InteractiveBrowserCredentialOptions extends IdentityClientOptions {
+    loginStyle?: BrowserLoginStyle;
+    postLogoutRedirectUri?: string | (() => string);
+    redirectUri?: string | (() => string);
+}
+
+// @public
+export class ManagedIdentityCredential implements TokenCredential {
+    constructor(clientId?: string, options?: IdentityClientOptions);
```

Comment:
should clientId be part of the options bag since it's optional?

Reply:
Yes.

Reply:
in this case it should be overload so we can doc user assigned identity